File: frontend/src/types/cvScreening.ts
export interface LinkedInAccount {
  name: string;
  label: string;
}

export interface LinkedInPostResult {
  account: string;
  label: string | null;
  authorUrn: string | null;
  postUrn: string | null;
  postUrl: string | null;
  postedAt: string | null;
  error: string | null;
}

export interface JobDescription {
  id: number;
  title: string;
  departmentId: number;
  descriptionText: string;
  requirementsText: string | null;
  filePath: string | null;
  status: "draft" | "open" | "closed" | string;
  createdBy: number;
  createdAt: string;
  updatedAt: string;
  applicantsCount: number;
  applicationFormUrl: string | null;
  linkedinPosts: LinkedInPostResult[];
}

export interface JobDescriptionCreate {
  title: string;
  departmentId: number;
  descriptionText: string;
  requirementsText?: string;
  status?: "draft" | "open" | "closed";
  linkedinAccountNames?: string[];
}

export interface JobPostingAiDraftRequest {
  title: string;
  departmentId: number;
}

export interface JobPostingAiDraftSkill {
  name: string;
  level: number;
}

export interface JobPostingAiDraftResult {
  descriptionText: string;
  requirementsText: string;
  skills: JobPostingAiDraftSkill[];
  applicationFormUrl: string | null;
}

export interface ScoringCriteria {
  id: number;
  jobDescriptionId: number;
  criterionName: string;
  weight: number;
  scoringRules: Record<string, unknown> | null;
  createdAt: string;
  updatedAt: string;
}

export interface ScoringCriteriaInput {
  criterionName: string;
  weight: number;
  scoringRules: Record<string, unknown>;
}

export type CvSource = "manual" | "webmail" | "outlook" | "whatsapp" | "gmail" | "google_form";

export interface Candidate {
  id: number;
  jobDescriptionId: number | null;
  fullName: string | null;
  email: string | null;
  phone: string | null;
  cvFilePath: string;
  parsedData: Record<string, unknown> | null;
  status: string;
  source: CvSource;
  sourceRef: string | null;
  matchConfidence: number | null;
  matchReasoning: string | null;
  submittedAt: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface RankingRow {
  candidateId: number;
  fullName: string | null;
  email: string | null;
  status: string;
  totalScore: number;
  rankPosition: number;
  pendingManualReview: boolean;
}

export interface SkillEvaluationRow {
  skill: string;
  requiredLevel: number;
  matched: boolean;
  rawScore: number | null;
  maxPoints: number;
  notes: string | null;
}

export interface CandidateEvaluation {
  candidateId: number;
  jobDescriptionId: number;
  jobTitle: string;
  overallScore: number | null;
  rankPosition: number | null;
  ratingOutOf10: number | null;
  recommendation: "shortlist" | "consider" | "reject";
  recommendationLabel: string;
  summary: string;
  whyAccepted: string;
  whyRejected: string;
  strengths: string[];
  gaps: string[];
  skills: SkillEvaluationRow[];
}

export interface CvSourceResult {
  source: "webmail" | "outlook" | "whatsapp" | "gmail" | "google_form";
  configured: boolean;
  fetched: number;
  message: string | null;
}

export interface CvSyncResult {
  sources: CvSourceResult[];
  totalFetched: number;
  autoMatched: number;
  unassigned: number;
  duplicatesSkipped: number;
  restoredFiles?: number;
  candidates: Candidate[];
}

export interface CandidateAssignRequest {
  jobDescriptionId: number;
}

File: backend/app/schemas/cv_screening.py
"""CV screening / candidate schemas."""
from __future__ import annotations

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field


class CandidateCreateMeta(BaseModel):
    full_name: str | None = None
    email: str | None = None
    phone: str | None = None


class CandidateUpdate(BaseModel):
    full_name: str | None = None
    email: str | None = None
    phone: str | None = None
    parsed_data: dict[str, Any] | None = None
    status: Literal["uploaded", "parsed", "scored", "shortlisted", "rejected", "hired"] | None = None


class CandidateRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    job_description_id: int
    full_name: str | None
    email: str | None
    phone: str | None
    cv_file_path: str
    parsed_data: dict[str, Any] | None
    status: str
    created_at: datetime
    updated_at: datetime


class CandidateScoreRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    candidate_id: int
    scoring_criteria_id: int
    raw_score: float | None
    notes: str | None


class CandidateRankingRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    job_description_id: int
    candidate_id: int
    total_score: float
    rank_position: int
    computed_at: datetime


class RankingRow(BaseModel):
    candidate_id: int
    full_name: str | None
    email: str | None
    status: str
    total_score: float
    rank_position: int
    pending_manual_review: bool = False


class ScoreOverrideRequest(BaseModel):
    scoring_criteria_id: int
    raw_score: float
    reason: str = Field(min_length=3)


class SkillEvaluationRow(BaseModel):
    skill: str
    required_level: float
    matched: bool
    raw_score: float | None
    max_points: float
    notes: str | None


class CandidateEvaluation(BaseModel):
    candidate_id: int
    job_description_id: int
    job_title: str
    overall_score: float | None
    rank_position: int | None
    recommendation: Literal["shortlist", "consider", "reject"]
    recommendation_label: str
    summary: str
    strengths: list[str]
    gaps: list[str]
    skills: list[SkillEvaluationRow]


class HireRequest(BaseModel):
    employee_code: str
    department_id: int
    role_title: str | None = None
    base_salary: float | None = None
    date_joined: str | None = None  # ISO date
